src/ui/public/state_management/state.js

```diff
@@ -205,7 +205,11 @@ export function StateProvider(Private, $rootScope, $location, config, kbnUrl) {
    */
   State.prototype._parseQueryParamValue = function (queryParam) {
     if (!isStateHash(queryParam)) {
```

Comment:
__question:__ There is __probably__ another way to fix this issue: just remove this `decode` code from here entirely. I've noticed that it's used only in two places:

1. https://github.com/elastic/kibana/blob/master/src/ui/public/state_management/state.js#L73-L75. Here `isStateHash` check is performed outside of this method, so `decode` part is not used;
2. https://github.com/elastic/kibana/blob/master/src/ui/public/state_management/state.js#L227. And here for non-hash case we do decode->encode that seems unnecessary.

So if there is no any non-obvious value in `decode->encode` cycle, we can probably just add `isStateHash` check inside 2. and remove `decode` part from this method.

How does that sound?